tests/vmi_bridge_test.go

```diff
@@ -0,0 +1,480 @@
+							Volumes: []k8sv1.Volume{
+								{
+									Name: "var-run", VolumeSource: k8sv1.VolumeSource{
+										HostPath: &k8sv1.HostPathVolumeSource{Path: "/var/run"}},
+								},
+								{
+									Name: "device-plugin", VolumeSource: k8sv1.VolumeSource{
+										HostPath: &k8sv1.HostPathVolumeSource{Path: "/var/lib/kubelet/device-plugins"}},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			_, err = virtClient.AppsV1().DaemonSets(tests.NamespaceTestDefault).Create(&ds)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		waitForPodToFinish := func(pod *k8sv1.Pod) k8sv1.PodPhase {
+			Eventually(func() k8sv1.PodPhase {
+				j, err := virtClient.Core().Pods(tests.NamespaceTestDefault).Get(pod.ObjectMeta.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return j.Status.Phase
+			}, 60*time.Second, 1*time.Second).Should(Or(Equal(k8sv1.PodSucceeded), Equal(k8sv1.PodFailed)))
+			j, err := virtClient.Core().Pods(tests.NamespaceTestDefault).Get(pod.ObjectMeta.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return j.Status.Phase
+		}
+
+		addBridgeToHost := func(name string) {
+			// create bridge on the node
+			parameters := []string{"link", "add", name, "type", "bridge"}
+			job := tests.RenderIPRouteJob(fmt.Sprintf("ip-add-%s", name), parameters)
+
+			// make sure that both jobs are happening on the same node
+			listOptions := metav1.ListOptions{}
+			nodeList, err := virtClient.CoreV1().Nodes().List(listOptions)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(nodeList.Items).NotTo(HaveLen(0))
+			node := nodeList.Items[0]
+			nodeSelector := map[string]string{"kubernetes.io/hostname": node.Name}
+
+			job.Spec.NodeSelector = nodeSelector
+			job, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(job)
+			Expect(err).ToNot(HaveOccurred())
+			waitForPodToFinish(job)
+			// dont check results, as this may fail because bridge is already there
+			// if there was any issue with creating the bridges the following "set" command would indicate the failure
+
+			// set the bridge to "up" mode
+			parameters = []string{"link", "set", "dev", name, "up"}
+			job = tests.RenderIPRouteJob(fmt.Sprintf("ip-set-%s", name), parameters)
+			job.Spec.NodeSelector = nodeSelector
+			job, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(job)
+			Expect(err).ToNot(HaveOccurred())
+			phase := waitForPodToFinish(job)
+			Expect(phase).To(Equal(k8sv1.PodSucceeded))
+
+			// set IP on bridge
+			parameters = []string{"addr", "add", bridgeIP[name] + "/24", "dev", name}
+			job = tests.RenderIPRouteJob(fmt.Sprintf("ip-addr-%s", name), parameters)
+			job.Spec.NodeSelector = nodeSelector
+			job, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(job)
+			Expect(err).ToNot(HaveOccurred())
+			phase = waitForPodToFinish(job)
+			// don't check address setting result
+		}
+
+		// add red and blue bridges to host
+		addBridgeToHost("red")
+		addBridgeToHost("blue")
```

Comment:
I have a small question, do we need to remove the bridges after finishing the testing?